app/javascript/components/CountrySelectionModal.tsx

```diff
@@ -50,11 +50,11 @@ export const CountrySelectionModal = ({ country: initialCountry, countries }: Pr
       <Modal
         open
         onClose={() => {
-          const referrer = document.referrer;
-          if (referrer && new URL(referrer).origin === window.location.origin) {
-            window.history.back();
+          const previousRoute = sessionStorage.getItem("inertia_previous_route");
```

Comment:
we extract the previous path from the session storage as set by the inertia middleware